draft-tuexen-opsawg-pcapng.xml

```diff
@@ -2090,15 +2090,21 @@ Section Header
 
             <t hangText="0x5a4e574b:"><vspace blankLines="0"/>ZigBee NWK Key.
             Network Key as described in the <eref target="https://zigbeealliance.org/">
-            ZigBee Specification</eref> R21 section 4.2.2. The NWK Key is a 16 octet
-            AES-128 key used to secure NWK Level frames within a single PAN.
+            ZigBee Specification</eref> 05-3473-21 (R21) section 4.2.2. The NWK Key
+            is a 16 octet binary AES-128 key used to secure NWK Level frames
+            within a single PAN. If and when the NWK Key changes a new DSB will
+            contain the new NWK Key.
             </t>
 
             <t hangText="0x5a415053:"><vspace blankLines="0"/>ZigBee APS Key.
             Application Support Link Key as described in the 
             <eref target="https://zigbeealliance.org/"> ZigBee Specification</eref>
-            R21 sections 2.2.5.1 and 4.4. Each 16 octet AES-128 key secures
-            frames exchanged between a pair of network nodes.
+            05-3473-21 (R21) section 4.4. Each 16 octet binary AES-128 key secures
+            frames exchanged between a pair of network nodes. There is a APS
+            Key DSB for each node pair for which the Link Key is known. As new links
+            are formed, new DSBs contain the new Keys. If the APS Key changes
+            for an existing link, it is contained in a new DSB with the new
+            APS Key.
```

Comment:
So an APS Key entry applies to all packets between the nodes in packet blocks between the DSB in which an APS Key appears and the next DSB in which an APS Key for that node pair appears?

> There is a APS Key DSB for each node pair for which the Link Key is known.

So does an APS Key entry include, in addition to the key, an indication of the node pair to which it applies?

Reply:
The APS Key will behave the same as the NWK Key above. Unfortunately the design of the dissector is such that both NWK keys and APS keys are stored in the same keyring. This means that APS Keys will also include the PANID. Although they are not needed now, we should probably include the short address of both pairs. Because the link key is symmetric the order doesn't matter (A,B) or (B,A) so to avoid having to check both, I would suggest a convention (low,high) i.e. (A,B) where A<B. I'll look at adding this change too.

Reply:
> the design of the dissector

...is irrelevant; pcapng isn't a tcpdump or Wireshark file format, it's a general file format, and if the best design forces tcpdump or Wireshark dissectors to be changed, so be it.

Reply:
This is why I've added separate tags for NWK and APS keys. The dissector doesn't distinguish, but from a ZigBee perspective it should.